docs: Update feed doc (#427)
unoplatform/uno.extensions: doc/Reactive/Reactive.06-in-apps.md

* An `IInput<T>` to get _data_ from the view (e.g. for 2-way bindings);
* An `ICommandBuilder` for the "trigger" inputs.

In order to easily interact with the binding engine in a performant way, a `BindableXXX` class is automatically generated. It is this class that will hold the state and which **has to been set as `DataContext` of your page**.

> Note: The bindable counterpart of a class will be created when the class name ends with "ViewModel".
> You can customize that behavior using the `[ReactiveBindable]` attribute.

> [!NOTE]
> For easier View Model creation, public properties will also be accessible on the `BindableXXX` class as long as there is no name conflict between these properties and any constructor's _input_. You can also access to the view model itself through the `Model` property.

## Display some data in the UI (VM to View)

To render a _feed_ in the UI, you only have to expose it through a public property:
```csharp
public IFeed<Product[]> Products = Feed.Async(_productService.GetProducts);
```

Then in your page, you can add a `FeedView`:

> ```xaml

Reviewer: while `xaml` snippet works here on github
it wont have any syntax highlighting once we fed it through docfx
case in point:
source: https://github.com/unoplatform/uno/blob/master/doc/articles/features/progressring.md?plain=1#L45
preview: https://unoplatformdocstaging.z13.web.core.windows.net/articles/features/progressring.html?q=progr#platform-specific-usage

Author: Good to know, but does using a XML tag would significantly improve the result? I think that we should still use "XAML" as it works on GitHub, and I would assume that it's something that can be fixed on our instance of docfx.

Reviewer: xml highlight is pretty much all you could ask for xaml
i would say we should just use \`\`\`xml, until xaml is fixed

Author: In my point of view, the "until xaml is fixed" is a synonym to "never" so I would prefer to keep it like that ... but as it seems we won't be able to have it in short time-frame and anyway the support of XAML on GitHub seems to be just an alias of XML, I'll change to XML :(

# Usage in applications of feeds

The recommended use for feeds is only in view models.

The reactive framework allows you to design state-less view models, focusing on the presentation logic. 
Your view models only have to request in their constructors the user _ inputs_ that are expected from the view. Inputs could be:
* An `IInput<T>` to get _data_ from the view (e.g. for 2-way bindings);
* An `ICommandBuilder` for the "trigger" inputs.

> In order to easily interact with the binding engine in a performant way, a `BindableXXX` class is automatically generated. It is this class that will hold the state and which **has to been set as `DataContext` of your page**.

Reviewer 2: ```suggestion
In order to easily interact with the binding engine in a performant way, a `BindableXXX` class is automatically generated. It is this class that will hold the state and which **has to be set as `DataContext` of your page**.
```

> [!NOTE]
> The bindable counterpart of a class will be created when the class name ends with "ViewModel".
> You can customize that behavior using the `[ReactiveBindable]` attribute.

> [!NOTE]
> For easier View Model creation, public properties will also be accessible on the `BindableXXX` class. 

> > You can also access to the view model itself through the `Model` property.

Reviewer 2: ```suggestion
> You can also access the view model itself through the `Model` property.
```

## Display some data in the UI (VM to View)

To render a _feed_ in the UI, you only have to expose it through a public property:
```csharp
public IFeed<Product[]> Products = Feed.Async(_productService.GetProducts);
```

Then in your page, you can add a `FeedView`:
```xml
<Page 
	x:Class="MyProject.MyPage"
	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	xmlns:uer="using:Uno.Extensions.Reactive.UI">

<uer:FeedView Source="{Binding Products}">
	<DataTemplate>
		<ListView ItemsSource="{Binding Data}" />
	</DataTemplate>
</uer:FeedView>
```

## Commands

> The generated bindable counterpart of a class will automaticlaly re-expose public methods that has 0 or 1 parameter and an optional `CancellationToken` as `ICommand`.

Reviewer 2: ```suggestion
The generated bindable counterpart of a class will automatically re-expose public methods that have 0 or 1 parameter and an optional `CancellationToken` as `ICommand`.
```
